### DISCUSSION TITLE
Generalization table: can i make a Prisma model relation?

### DISCUSSION THREAD

**Discussion by rpenido** [prisma/prisma#13033] [Q&A]
Hi!

I'm working on a project and came across the following pattern:
```prisma
model Vehicle {
  id        Int
  tableName String
  tableId   Int
  ...
}

model Car {
  id       Int
  ...
  carField String
}

model Bus {
  id       Int
  ...
  busField String
}
```

I have a generalized register for Vehicle and specialized for Car like
```javascript
Vehicle
[
  {
    id: 32,
    tableName: 'car',
    tableId: 13,
  },
  {
    id: 33,
    tableName: 'bus',
    tableId: 11,
  }
]

Car
{
 id: 13,
 carField: 'someValue',
}

Bus
{
 id: 11,
 busField: 'someOtherValue',
}
```

This pattern is all over the code.
I don't think that's a good design, but I need to work with this until I figure out a better solution and migrate all production data.

I'm quering `Vehicle` and want to aggregate data from Car/Bus.

There is a way to create a relation in this ugly pattern?
```prisma
model Vehicle {
  id        Int
  tableName String
  tableId   Int
  ...
  car       Car @relation(fields: [tableId], references: [id], where: {tableName: 'car'})
  bus       Bus @relation(fields: [tableId], references: [id], where: {tableName: 'bus'})
}
```
The only solution is rawQueries ?

I would like to hear from you guys a better approach (short and/or long term) to deal with this pattern!
Thank you!

**Comment by austincrim** [comment]
Hey @rpenido 👋 ,

There isn't anything built into Prisma to help make this migration. You will need to migrate existing data to the new data model but use `$rawQuery` in the meantime.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
